src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -3,6 +3,142 @@ Ceph Dashboard Developer Documentation
 
 .. contents:: Table of Contents
 
+Preliminary Steps
+-----------------
+
+The following documentation chapters expect a running Ceph cluster and at
+least a running `dashboard` manager module (with few exceptions). This
+chapter gives an introduction on how to set up such a system for development,
+without the need to set up a full blown production environment. All in this
+chapter introduced options are based on a so called `vstart` environment. 
+
+.. note::
+
+  Every `vstart` environment needs Ceph `to be compiled`_ from its Github
+  repository, though Docker environments simplify that step by providing a
+  shell script that contains those instructions.
```

Comment:
Good point. I think it makes sense to point out both options and available solutions, in order to not discriminate one approach or the other. I think there is merit in some diversity/choice here.

Reply:
> if this is mostly meant for dashboard newcomers/interns

Indeed, it is. But for completeness, I wouldn't leave out the host option, but of course that's also not what you've asked for.

I like to hear that there's a build-free solution available and I'm glad to include it in the doc! Also thanks for directly pointing me to it, as I would have had to search for it otherwise!

Reply:
..but can you briefly explain the benefit for development? It came to my mind that it might lack a proper Github repository. Does it has other downsides?

Reply:
Just that I'll be able to put it in correct term. I don't want to recommend a development environment which turns out not to be one, because one cannot keep track of any changes or one is unable to commit and push them. If it's just about getting the dashboard running to have a look and being able to click around, I've got to put that into the proper perspective.

Reply:
> Just that I'll be able to put it in correct term. I don't want to recommend a development environment which turns out not to be one

@p-se the build-free solution is also a development environment, you can make changes
as all the mgr code is shared  with your local repo, so you can perfectly work with it (the only scenario where would not help is when you need to edit C++ code, as in that case you're using rpm packages).
In fact, it has been like a year without having to build in order to send PRs.

Reply:
@alfonsomthd 
Can you share the build-free solution? Link, or guidelines. Thanks!

Reply:
@p-se I think Alfonso already replied (thanks @alfonsomthd !).

@bk201 you can follow [these steps](https://github.com/rhcs-dashboard/ceph-dev#quick-install-rpm-based)(`ceph-dev` only). I'm sure it won't be really complicate to bring that feature into `ceph-dev-docker` too (as `ceph-dev` was originally based on `ceph-dev-docker`).

Reply:
That's quite a good selling point